### PR TITLE
fixes runtime panic in state/getters

### PR DESCRIPTION
**Pull request by farazdagi**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**
- When running sync, got runtime panic, due to nil state dereference. This PR adds non-nil checks.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/state/getters.go [hunk 1]
```diff
@@ -450,13 +450,19 @@ func (b *BeaconState) ValidatorAtIndexReadOnly(idx uint64) (*ReadOnlyValidator,
...
+		return 0, false
+	}
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 	idx, ok := b.valIdxMap[key]
 	return idx, ok
 }
 
 func (b *BeaconState) validatorIndexMap() map[[48]byte]uint64 {
+	if b == nil || b.valIdxMap == nil {
```
Comment (nisdas) [inline beacon-chain/state/getters.go:463]: this is a bit dangerous. The caller of this method will get a nil map, which will definitely lead to a panic when they try to read from that map.

Comment (farazdagi) [reply beacon-chain/state/getters.go:463]: I see, so short of rewriting this method to return error (or book ok), what can we do?

Comment (farazdagi) [reply beacon-chain/state/getters.go:463]: Should I return inited empty map then?

Comment (nisdas) [reply beacon-chain/state/getters.go:463]: I think an empty map should be fine here instead

Comment (farazdagi) [reply beacon-chain/state/getters.go:463]: Done! Thank you for noticing this.

**Pull request merged**